python/ml4ir/base/pipeline.py

```diff
@@ -396,19 +397,35 @@ def get_relevance_model(self, feature_layer_keys_to_fns={}) -> RelevanceModel:
             aux_metrics = self.get_metrics(self.aux_metrics_keys)
 
         # Define scorer
-        scorer: RelevanceScorer = RelevanceScorer(
-            feature_config=self.feature_config,
-            model_config=self.model_config,
-            interaction_model=interaction_model,
-            loss=loss,
-            aux_loss=aux_loss,
-            aux_loss_weight=self.args.aux_loss_weight,
-            aux_metrics=aux_metrics,
-            output_name=self.args.output_name,
-            logger=self.logger,
-            file_io=self.file_io,
-            logs_dir=self.logs_dir_local
-        )
+        if ("monte_carlo_inference_trials" in self.model_config and
+                self.model_config["monte_carlo_inference_trials"].get("value", 0)):
+            scorer: RelevanceScorer = MonteCarloScorer(
+                feature_config=self.feature_config,
+                model_config=self.model_config,
+                interaction_model=interaction_model,
+                loss=loss,
+                aux_loss=aux_loss,
+                aux_loss_weight=self.args.aux_loss_weight,
+                aux_metrics=aux_metrics,
+                output_name=self.args.output_name,
+                logger=self.logger,
+                file_io=self.file_io,
+                logs_dir=self.logs_dir_local
+            )
+        else:
+            scorer: RelevanceScorer = RelevanceScorer(
+                feature_config=self.feature_config,
+                model_config=self.model_config,
+                interaction_model=interaction_model,
+                loss=loss,
+                aux_loss=aux_loss,
+                aux_loss_weight=self.args.aux_loss_weight,
+                aux_metrics=aux_metrics,
```

Comment:
@mohazahran I had a comment about moving this to a separate file under `scorer/` like `scorer_factory.py` which has a `get_scorer` method to keep this pipeline clean. We've bloated it quite a bit already. Want to avoid further.